### ISSUE TITLE
Build of 0.3.0 failed

### ISSUE THREAD

**Issue by fhaefemeier** [mozilla/sccache#1170]
Compiling sccache 0.3.0 with latest rust version 1.60 is failing with the following error message:
```
Compiling sccache v0.3.0
error[E0599]: no method named `to_rfc7231` found for struct `chrono::DateTime` in the current scope
   --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/sccache-0.3.0/src/simples3/s3.rs:90:51
    |
90  |             let date = chrono::offset::Utc::now().to_rfc7231();
    |                                                   ^^^^^^^^^^ help: there is an associated function with a similar name: `to_rfc3339`
    |
    = help: items from traits can only be used if the trait is implemented and in scope
note: `DateTimeExt` defines an item `to_rfc7231`, perhaps you need to implement it
   --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/sccache-0.3.0/src/util.rs:510:1
    |
510 | pub trait DateTimeExt {
    | ^^^^^^^^^^^^^^^^^^^^^
error[E0599]: no method named `to_rfc7231` found for struct `chrono::DateTime` in the current scope
   --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/sccache-0.3.0/src/simples3/s3.rs:124:47
    |
124 |         let date = chrono::offset::Utc::now().to_rfc7231();
    |                                               ^^^^^^^^^^ help: there is an associated function with a similar name: `to_rfc3339`
    |
    = help: items from traits can only be used if the trait is implemented and in scope
note: `DateTimeExt` defines an item `to_rfc7231`, perhaps you need to implement it
   --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/sccache-0.3.0/src/util.rs:510:1
    |
510 | pub trait DateTimeExt {
    | ^^^^^^^^^^^^^^^^^^^^^
For more information about this error, try `rustc --explain E0599`.
error: failed to compile `sccache v0.3.0`, intermediate artifacts can be found at `/tmp/cargo-installDFqR4j`
Caused by:
  could not compile `sccache` due to 2 previous errors
```

**Comment by fhaefemeier** [comment]
Any news. It is still not possible to compile from source even with latest rust version 1.61

**Comment by sylvestre** [comment]
No news yet, don't hesitate to propose a fix!

**Comment by fhaefemeier** [comment]
@sylvestre thanks for your answer. But I am wondering, how is it possible to compile this version by your own or the community? The error message indicates a general problem with the chrono library. Regarding to the compile log it use the latest version 0.4.19. I want to create a sccache version with S3 enabled.
To install sccache in my system I call 
```
cargo install --no-default-features --features=s3 sccache
```

**Comment by Xanewok** [comment]
Could you paste the error log as well?

**Comment by glandium** [comment]
Add --locked to your `cargo install` command.

**Comment by glandium** [comment]
```
cargo install --no-default-features --features=s3,azure sccache
```
works. There must be some problem in feature-gated things.

**Comment by fhaefemeier** [comment]
@glandium I can reproduce it. With s3 and azure enabled feature, it worked. `--locked` has no effect.
@Xanewok There is no more as the log in my first comment. But nevertheless I attached a [log](https://github.com/mozilla/sccache/files/8840774/sccache-0.3.0.log) from the docker run mentioned below.

With
```
docker run --rm -it rust:1.61 cargo install --no-default-features --features=s3 sccache
```
you will get a failed build. with an addtional `azure` feature enabled, it works.
The environment is fresh and no interference from the host system.

**Comment by Ten0** [comment]
This appears to be fixed by e6f72f21553ec76dc3b1b3cfd9f4d2c787c9f821, which itself appears to not be released.
